components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -158,26 +154,17 @@ public void activateOptions() {
     }
 
     private void publisherInitializer() {
-        Properties passwordProperty = new Properties();
-        passwordProperty.put("log4j.appender.DAS_AGENT.password", password);
-        SecretResolver secretResolver = SecretResolverFactory.create(passwordProperty);
-        String secretAlias = "log4j.appender.DAS_AGENT.password";
-        //Checking the log4j appender DAS credentials.
-        if (secretResolver != null && secretResolver.isInitialized()) {
-            if (secretResolver.isTokenProtected(secretAlias)) {
-                password = secretResolver.resolve(secretAlias);
-            } else {
-                password = (String) passwordProperty.get(secretAlias);
-            }
-        }
+        resolveSecretPassword();
         try {
-            dataPublisher = new DataPublisher("Thrift", url, authURLs, userName, password);
+            dataPublisher = new DataPublisher(protocol, url, authURLs, userName, password);
         } catch (DataEndpointAgentConfigurationException e) {
-            LogLog.error("Invalid urls passed for receiver and auth, and hence expected to fail " + e.getMessage(), e);
+            LogLog.error("Invalid urls passed for receiver and auth, and hence expected to fail for data agent " + e
```

Comment:
This is fixed in following commit : https://github.com/wso2/shared-analytics/pull/140/commits/1552e3cd174827a3d43c47e56938572cc11fa7b2